builtin/logical/aws/path_static_roles.go

```diff
@@ -0,0 +1,278 @@
+	// Create & validate config from request parameters
+	config := staticRoleEntry{}
+
+	if rawRoleName, ok := data.GetOk(paramRoleName); ok {
+		config.Name = rawRoleName.(string)
+
+		if err := b.validateRoleName(config.Name); err != nil {
+			return nil, err
+		}
+	} else {
+		return nil, fmt.Errorf("missing '%s' parameter", paramRoleName)
+	}
+
+	if rawUsername, ok := data.GetOk(paramUsername); ok {
+		config.Username = rawUsername.(string)
+
+		if err := b.validateIAMUserExists(ctx, req, config.Username); err != nil {
+			return nil, err
+		}
+	} else {
+		return nil, fmt.Errorf("missing '%s' parameter", paramUsername)
+	}
+
+	if rawRotationPeriod, ok := data.GetOk(paramRotationPeriod); ok {
+		config.RotationPeriod = time.Duration(rawRotationPeriod.(int)) * time.Second
+
+		if err := b.validateRotationPeriod(config.RotationPeriod); err != nil {
+			return nil, err
+		}
+	} else {
+		return nil, fmt.Errorf("missing %q parameter", paramRotationPeriod)
+	}
+
+	b.roleMutex.Lock()
+	defer b.roleMutex.Unlock()
+
+	// Upsert role config
+	newRole, err := logical.StorageEntryJSON(formatRoleStoragePath(config.Name), config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal object to JSON: %w", err)
+	}
+	err = req.Storage.Put(ctx, newRole)
+	if err != nil {
+		return nil, fmt.Errorf("failed to save object in storage: %w", err)
+	}
+
+	// Bootstrap initial set of keys if they did not exist before. AWS Secret Access Keys can only be obtained on creation,
+	// so we need to boostrap new roles with a new initial set of keys to be able to serve valid credentials to Vault clients.
+	existingCreds, err := req.Storage.Get(ctx, formatCredsStoragePath(config.Name))
+	if err != nil {
+		return nil, fmt.Errorf("unable to verify if credentials already exist for role %q: %w", config.Name, err)
+	}
+	if existingCreds == nil {
+		err := b.createCredential(ctx, req.Storage, config, false)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create new credentials for role %q: %w", config.Name, err)
+		}
+
+		err = b.credRotationQueue.Push(&queue.Item{
+			Key:      config.Name,
+			Value:    config,
+			Priority: time.Now().Add(config.RotationPeriod).Unix(),
+		})
+		if err != nil {
+			return nil, fmt.Errorf("failed to add item into the rotation queue for role %q: %w", config.Name, err)
+		}
+	}
+
+	return &logical.Response{
+		Data: formatResponse(config),
+	}, nil
+}
+
+func (b *backend) pathStaticRolesDelete(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	roleName, ok := data.GetOk(paramRoleName)
+	if !ok {
+		return nil, fmt.Errorf("missing '%s' parameter", paramRoleName)
+	}
+
+	err := b.deleteCredential(ctx, req.Storage, roleName.(string), true)
+	if err != nil {
+		return nil, fmt.Errorf("failed to clean credentials while deleting role %q: %w", roleName.(string), err)
+	}
+
+	return nil, req.Storage.Delete(ctx, formatRoleStoragePath(roleName.(string)))
```

Comment:
Should we be removing this item from the queue before we delete it?

Reply:
I think we either should or we should check to see if the role still exists when we go to rotate.

Reply:
There's also the question of the access key itself - i suppose we should clean up everything we can unless someone has an argument not to. We're already deleting the access key that we know about each rotation.

Reply:
Should we check if it exists in the queue before we rotate in either case? Are there any other consequences if we don't remove it from the queue here?

Reply:
rotation should only happen via the queue - the periodic function peeks at the top element of the queue, decides if the referenced credential needs rotation, and does it, repeating until it hits one that doesn't need rotation.

Reply:
(aside from the initial "rotation" when the role is first created)